### ISSUE TITLE
Deno.dlopen cannot open static asset

### ISSUE THREAD

**Issue by vseplet** [denoland/deno#27068]
Version: Deno 2.1.1

I tried to bundle my native library and script into a single executable file. Of course, I used the --include flag during compilation. After running the executable, I found that Deno.readFile successfully reads the static asset, but Deno.dlopen is unable to open it. Below is an example of the code:

```ts
const data = Deno.readFileSync(
  import.meta.dirname + "/static/libexample.dylib",
);

console.log(data.buffer.byteLength); // return correct size 1955522

const lib = Deno.dlopen(
  import.meta.dirname + "/static/libexample.dylib",
  {
    openWebView: {
      parameters: [],
      result: "void",
      nonblocking: false,
    },
  },
); // but Error: Could not open library: Could not open library

```

**Comment by eibens** [comment]
Weirdly enough, if you use `const tmp = await Deno.makeTempFile()` and copy the library file to it, you can use `Deno.dlopen(tmp)` without issues.

**Comment by sigmaSd** [comment]
@eibens yes that's the current workaround, it's to bundle the native library and copy it to the host system before using it

**Comment by sigmaSd** [comment]
This is should be fixed in 2.3

**Comment by vseplet** [comment]
@sigmaSd Thank you very much for your work!
